paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -196,15 +211,47 @@ internal abstract class BaseSheetViewModel<TransitionTargetType>(
         }
     }
 
-    fun getSupportedPaymentMethods(): List<SupportedPaymentMethod> {
-        stripeIntent.value?.let { stripeIntent ->
-            return stripeIntent.paymentMethodTypes.mapNotNull {
+    private fun getSupportedPaymentMethods(
+        stripeIntentParameter: StripeIntent?
+    ): List<SupportedPaymentMethod> {
+        stripeIntentParameter?.let { stripeIntent ->
+            return stripeIntent.paymentMethodTypes.asSequence().mapNotNull {
                 SupportedPaymentMethod.fromCode(it)
             }.filterNot {
                 // AfterpayClearpay requires a shipping address, filter it out if not provided
-                it == SupportedPaymentMethod.AfterpayClearpay &&
+                val excludeAfterPay = it == SupportedPaymentMethod.AfterpayClearpay &&
                     (stripeIntent as? PaymentIntent)?.shipping == null
+                if (excludeAfterPay && BuildConfig.DEBUG) {
```

Comment:
The logger already takes care of logging at debug level only if this is a debug build, I don't think we need to add this check everywhere.